velox/exec/tests/TaskTest.cpp

```diff
@@ -1314,4 +1314,43 @@ TEST_F(TaskTest, driverCreationMemoryAllocationCheck) {
     }
   }
 }
+
+TEST_F(TaskTest, spillDirectoryLifecycleManagement) {
+  // Marks the spill directory as not already created and ensures that the Task
+  // handles creating it on first use and eventually deleting it on destruction.
+  auto data = makeRowVector({
+      makeFlatVector<int64_t>(1'000, [](auto row) { return row % 300; }),
+      makeFlatVector<int64_t>(1'000, [](auto row) { return row; }),
+  });
+
+  core::PlanNodeId aggrNodeId;
+  const auto plan = PlanBuilder()
+                        .values({data})
+                        .singleAggregation({"c0"}, {"sum(c1)"}, {})
+                        .capturePlanNodeId(aggrNodeId)
+                        .planNode();
+  CursorParameters params;
+  params.planNode = plan;
+  params.queryCtx = std::make_shared<core::QueryCtx>(driverExecutor_.get());
+  params.queryCtx->testingOverrideConfigUnsafe(
+      {{core::QueryConfig::kSpillEnabled, "true"},
+       {core::QueryConfig::kAggregationSpillEnabled, "true"},
+       {core::QueryConfig::kTestingSpillPct, "100"}});
+  params.maxDrivers = 1;
+
+  auto cursor = std::make_unique<TaskCursor>(params);
+  std::shared_ptr<Task> task = cursor->task();
+  task->setSpillDirectory(
+      "/tmp/velox_test_spillDirectoryLifecycleManagement", false);
```

Comment:
Can we use createTempDirectory? and attach a suffix to make sure it is portal? Thanks!